test/unit/org/apache/cassandra/schema/TransactionalSchemaTest.java

```diff
@@ -0,0 +1,104 @@
+        SchemaTestUtil.addOrUpdateKeyspace(KeyspaceMetadata.create(KEYSPACE, KeyspaceParams.simple(1), Tables.of()));
+    }
+
+    private static void process(String fmt, Object... objects)
+    {
+        QueryProcessor.process(format(fmt, objects), ConsistencyLevel.ANY);
+    }
+
+    private static void assertTransactionalMode(String table, TransactionalMode mode, TransactionalMigrationFromMode migration)
+    {
+        TableMetadata metadata = Schema.instance.getTableMetadata(KEYSPACE, table);
+        Assert.assertEquals(mode, metadata.params.transactionalMode);
+        Assert.assertEquals(migration, metadata.params.transactionalMigrationFrom);
+    }
+
+    // if a table is created with an accord transactional mode, it skips having to migrate
+    @Test
+    public void newTableSkipsMigration()
+    {
+        String table = "new_table";
+        process("CREATE TABLE ks.%s (k int primary key, v int) WITH transactional_mode='%s'", table, TransactionalMode.full);
+        assertTransactionalMode(table, TransactionalMode.full, TransactionalMigrationFromMode.none);
+    }
+
+    // if an existing table is set to an accord transactional mode, it should be set to migrating
+    @Test
+    public void existingTableMigration()
+    {
+        String table = "existing_table";
+        process("CREATE TABLE ks.%s (k int primary key, v int)", table);
+        assertTransactionalMode(table, TransactionalMode.off, TransactionalMigrationFromMode.none);
+
+        process("ALTER TABLE ks.%s WITH transactional_mode='%s'", table, TransactionalMode.full);
+        assertTransactionalMode(table, TransactionalMode.full, TransactionalMigrationFromMode.off);
+    }
+
+    // changing transactional mode with an incomplete migration should fail, unless the migration mode is explicitly updated
+    @Test
+    public void incompleteMigrationFailure()
+    {
+        String table = "incomplete_table";
+        process("CREATE TABLE ks.%s (k int primary key, v int) WITH transactional_mode='%s' AND transactional_migration_from='%s'",
+                table, TransactionalMode.full, TransactionalMigrationFromMode.off);
+        assertTransactionalMode(table, TransactionalMode.full, TransactionalMigrationFromMode.off);
+
+        try
+        {
+            process("ALTER TABLE ks.%s WITH transactional_mode='%s'", table, TransactionalMode.off);
+            Assert.fail("Failure expected");
+        }
+        catch (Throwable t)
+        {
+            Assert.assertTrue(t instanceof InvalidRequestException);
+        }
+        assertTransactionalMode(table, TransactionalMode.full, TransactionalMigrationFromMode.off);
+
+        // explicitly setting the migration mode should work
+        process("ALTER TABLE ks.%s WITH transactional_mode='%s' AND transactional_migration_from='%s'",
+                table, TransactionalMode.off, TransactionalMigrationFromMode.none);
+        assertTransactionalMode(table, TransactionalMode.off, TransactionalMigrationFromMode.none);
```

Comment:
A lot of this doesn't really validate that it connects to the rest of the request routing stuff correctly. Maybe `MigrationTest` should be parameterized so that it can use `ALTER TABLE` where possible?